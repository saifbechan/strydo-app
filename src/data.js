import uuidv1 from 'uuid/v1';
import { LoremIpsum } from 'lorem-ipsum';
import randomColor from 'randomcolor';

const lorem = new LoremIpsum({
  wordsPerSentence: {
    max: 6,
    min: 3
  }
});

export default [
  {
    id: uuidv1(),
    title: 'Backlog',
    backgroundColor: randomColor(),
    cards: [
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      }
    ]
  },
  {
    id: uuidv1(),
    title: 'To do',
    backgroundColor: randomColor(),
    cards: [
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      }
    ]
  },
  {
    id: uuidv1(),
    title: 'In progress',
    backgroundColor: randomColor(),
    cards: [
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      }
    ]
  },
  {
    id: uuidv1(),
    title: 'Test',
    backgroundColor: randomColor(),
    cards: [
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      }
    ]
  },
  {
    id: uuidv1(),
    title: 'Done',
    backgroundColor: randomColor(),
    cards: [
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      },
      {
        id: uuidv1(),
        content: lorem.generateSentences(2)
      }
    ]
  }
];
