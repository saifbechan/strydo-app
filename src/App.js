import React from 'react';

import data from './data';

import './App.css';

import Column from './components/column/column.component';

function App() {
  return (
    <div className='App'>
      {data.map(column => (
        <Column key={column.id} {...column} />
      ))}
    </div>
  );
}

export default App;
