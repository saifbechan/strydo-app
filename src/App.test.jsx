import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import App from './App';

import data from './data';

jest.mock('./data');

it('renders correctly when there is no data', () => {
  data.mockImplementation(() => []);

  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders correctly when there is no data', () => {
  data.mockImplementation(() => [
    {
      id: '123-abc',
      title: 'xyz',
      backgroundColor: '#FFF',
      cards: []
    }
  ]);

  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it('adds a card when the `add-card-handler` is clicked', () => {
  data.mockImplementation(() => [
    {
      id: '123-abc',
      title: 'xyz',
      backgroundColor: '#FFF',
      cards: [
        {
          id: 'abc-123',
          content: 'content'
        }
      ]
    },
    {
      id: '456-def',
      title: 'nop',
      backgroundColor: '#000',
      cards: []
    }
  ]);

  const { getAllByTestId } = render(<App />);

  fireEvent.click(getAllByTestId('add-card-handler')[0]);
  expect(getAllByTestId('card')).toHaveLength(2);

  fireEvent.click(getAllByTestId('add-card-handler')[0]);
  expect(getAllByTestId('card')).toHaveLength(3);
});
