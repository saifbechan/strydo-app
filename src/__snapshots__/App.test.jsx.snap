// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is no data 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(240px,1fr));
  grid-template-rows: repeat(2,200px);
  grid-gap: 15px;
  margin: 0 15px;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`renders correctly when there is no data 2`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(240px,1fr));
  grid-template-rows: repeat(2,200px);
  grid-gap: 15px;
  margin: 0 15px;
}

.c1 {
  border: 1px solid grey;
}

.c2 {
  background-color: #FFF;
  font-size: 15px;
  margin: 0;
  padding: 10px;
}

.c3 {
  border-top: 1px solid grey;
}

.c4 {
  cursor: pointer;
  padding: 5px;
  text-align: center;
  width: 100%;
}

<div
    class="c0"
  >
    <div>
      <div
        class="c1"
        id="123-abc"
      >
        <h2
          class="c2"
        >
          xyz
        </h2>
        <div
          class="c3"
        />
      </div>
      <button
        class="c4"
        data-testid="add-card-handler"
      >
        + add card
      </button>
    </div>
  </div>
</DocumentFragment>
`;
